src/transform/tests/testdata/lifting

```diff
@@ -0,0 +1,195 @@
+# Copyright Materialize, Inc. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+cat
+(defsource x [int32 int64])
+----
+ok
+
+cat
+(defsource y [int64 int64])
+----
+ok
+
+# Discard literals that are not projected.
+build apply=LiteralLifting
+(project
+  (map (get x) [1 2 3 4])
+  [#5 #3])
+----
+%0 =
+| Get x (u0)
+| Map 2, 4
+| Project (#3, #2)
+
+
+# Merge nested Map operators within a Project
```

Comment:
Apparently on "main" one can get a nested map that is merged in this branch:

In "main":

```
materialize=>  EXPLAIN SELECT DISTINCT CAST( 9 AS DOUBLE PRECISION) AS c1, a1 . f2 AS c2, 8 AS c3, CAST( COUNT ( DISTINCT a1 . f1 ) AS DOUBLE PRECISION ) AS agg1 , CAST( MAX ( DISTINCT 4 ) AS DOUBLE PRECISION ) AS agg2 FROM t2 AS a1 RIGHT JOIN t1 AS a2 ON ( a1 . f2 IS NOT NULL ) WHERE CAST( 0 AS DOUBLE PRECISION) IS  NULL OR NOT ( CAST( 1 AS DOUBLE PRECISION) IS  NULL ) GROUP BY 1 , 2 , 3 
;
...
 %8 =                                            +
 | Union %2 %7                                   +
 | Reduce group=(9, #1, 8)                       +
 | | agg count(distinct #0)                      +
 | | agg dummy(dummy)                            +
 | Map i64tof64(#3)                              +
 | Map 4                                         +
 | Project (#0..#2, #5, #6)                      +
```

In this branch:

```
 %8 =                                            +
 | Union %2 %7                                   +
 | Reduce group=(#1)                             +
 | | agg count(distinct #0)                      +
 | | agg dummy(dummy)                            +
 | Map i64tof64(#1), 9, 8, 4                     +
 | Project (#4, #0, #5, #3, #6)                  +
```

Maybe the transforms need to be applied in a different order, or repeatedly?

Reply:
I added the removal of the nested maps as part of this transformation because in several cases it result in the addition of them as in the following case:

```
query T multiline
explain
select c1, c1 + a from (select 1 as c1, x as c2, 3 as c3 from generate_series(1, 3) as x union all select 1, x, 3 from generate_series(5, 8) as x), t;
----
%0 =
| Constant (1) (2) (3) (5) (6) (7) (8)

%1 =
| Get materialize.public.t (u1)

%2 =
| Join %0 %1
| | implementation = Differential %1 %0.()
| | demand = (#1)
| Map (1 + #1)
| Map 1
| Project (#4, #3)
```

I thought that since this transformation is now adding Map->Project constructions a lot, it was a good thing to absorb the nested ones in the projection.

Reply:
> Maybe the transforms need to be applied in a different order, or repeatedly?
This is interesting. I am going to try rebasing my PR to change the orders of the transforms (#6485) on this branch to see if multiple nested maps remain without the nested map absorption.

Reply:
It turns out that "main" produces nested maps because `LiteralLifting` runs as the last step in a `FixPoint` that runs after the last time `crate::fusion::map::Map` is called. To get rid of the nested maps, it suffices after add a  `crate::fusion::map::Map` after this line.
https://github.com/MaterializeInc/materialize/blob/51b785b14cc421eed3ca797915100a13c1412825/src/transform/src/lib.rs#L240